src/app/pages/LiquidityMining/hooks/useMining_ApproveAndRemoveLiquidityV1.ts

```diff
@@ -1,35 +1,32 @@
-import { Asset } from 'types/asset';
 import { getContract } from 'utils/blockchain/contract-helpers';
 import {
   CheckAndApproveResult,
   contractWriter,
 } from 'utils/sovryn/contract-writer';
 import { toWei } from 'utils/blockchain/math-helpers';
 import { useMining_RemoveLiquidityV1 } from './useMining_RemoveLiquidityV1';
-import { ContractName } from 'utils/types/contracts';
+import type { AmmLiquidityPool } from 'utils/models/amm-liquidity-pool';
 
 export function useMining_ApproveAndRemoveLiquidityV1(
-  pool: Asset,
+  pool: AmmLiquidityPool,
   amount: string,
-  reserveTokens: Asset[],
   reserveMinReturnAmounts: string[],
 ) {
   const { withdraw, ...txState } = useMining_RemoveLiquidityV1(
     pool,
     amount,
-    reserveTokens,
     reserveMinReturnAmounts,
   );
 
   return {
     withdraw: async () => {
       let tx: CheckAndApproveResult = {};
 
-      tx = await contractWriter.checkAndApproveContract(
-        `${pool}_${pool}_poolToken` as ContractName,
+      tx = await contractWriter.checkAndApproveAddresses(
+        pool.poolTokenA,
         getContract('BTCWrapperProxy').address,
         [amount, toWei('1000000000000000')],
```

Comment:
Why is the second amount hardcoded and what does this exactly mean? We should definitely have it in some local/global constant (depends if it's used elsewhere).

Reply:
Moved it to the constant.
When amount is an array of two items then function checks if token is approved for `amount` (first number) and if it's less than that, forces user to approve for second amount.
It basically overwrites default approval amount, cant remember why it had to be custom for LM without reading SC as it was made more than year ago